HW2/P4/driver.py

```diff
@@ -27,9 +28,29 @@ def py_median_3x3(image, iterations=10, num_threads=1):
 
     return tmpA
 
+def py_median_3x3(image, iterations=10, num_threads=4):
+    ''' repeatedly filter with a 3x3 median '''
+    tmpA = image.copy()
+    tmpB = np.empty_like(tmpA)
+    #Have each thread work on every num_thread row
+    for i in range(iterations):
+        threads = []
+        for nt in range(num_threads):
+            threads.append(threading.Thread(target=filtering.median_3x3,args=(tmpA, tmpB, nt, num_threads)))
```

Comment:
Okay, you locked each iteration in step. I think you could have used another communication primitive to do this.